Description: Add gap library directory
 Search libraries in gap's subdirectory where libgap is located.
Author: Tobias Hansen <thansen@debian.org>

--- a/sage/src/setup.py
+++ b/sage/src/setup.py
@@ -17,6 +17,7 @@
 from distutils.dep_util import newer_group
 from distutils.errors import (DistutilsSetupError, DistutilsModuleError,
                               DistutilsOptionError)
+import sysconfig
 
 # This import allows instancemethods to be pickable
 import fpickle_setup
@@ -87,7 +88,8 @@
 include_dirs = sage_include_directories(use_sources=True)
 
 # Look for libraries in $SAGE_LOCAL/lib
-library_dirs = [os.path.join(SAGE_LOCAL, "lib")]
+gap_library_dir = "/usr/lib/%s/gap" % sysconfig.get_config_var('MULTIARCH')
+library_dirs = [os.path.join(SAGE_LOCAL, "lib"), gap_library_dir]
 
 # Manually add -fno-strict-aliasing, which is needed to compile Cython
 # and disappears from the default flags if the user has set CFLAGS.
--- a/sage/src/sage/env.py
+++ b/sage/src/sage/env.py
@@ -293,7 +293,7 @@
 var('SINGULAR_SO', SINGULAR_SO)
 
 # locate libgap shared object
-GAP_SO= _get_shared_lib_filename('gap','')
+GAP_SO= "/usr/lib/%s/gap/libgap.so" % sysconfig.get_config_var('MULTIARCH')
 var('GAP_SO', GAP_SO)
 
 # post process
